src/App.tsx

```diff
@@ -7,17 +7,44 @@ import Header from './components/header/Header';
 import Content from './components/content/Content';
 
 import Home from './pages/home/Home';
-import NotFound from './pages/not-found/NotFound';
+import NotFound from './pages/404/404.tsx';
 import Term from './pages/term/Term';
+import { useEffect, useState } from 'preact/hooks';
 
 export default function App() {
+  const [sideNavOpen, setSideNavOpen] = useState(true);
```

Comment:
All navigation bar logic should be extracted to a different file as hook. ("useSideNavigation" or something)